### DISCUSSION TITLE
hydrateFoo() only gets called when changing the state of another field

### DISCUSSION THREAD

**Discussion by dev** [livewire/livewire#4780] [1. Help]
I have a registration form, with 4 fields:
- Name
- Email
- Password
- Password Confirmation

I want to transform the name using `Str::title()`, for example:  **john DOE** should become **John Doe**.
After seeing the Livewire screencasts, I could assume that I could use `hydrateName()` for this. For example:

```php
public function hydrateName($name) {
    $this->name = Str::title($name);
}
```

I tried this, but the name is only transformed to **John Doe** when the state of another field is changed.
Is this the intended behavior? What would be a better approach to achieve this?
In the screencasts, this did work the way I want it to.

**Comment by silnarm** [comment]
Not to sure exactly why your getting that behaviour, but it is a odd place to do what you want and that is likely why its not updating immediately.
Using the `updated` hook for this would be the normal route, change the hook function name to `updatedName` and see if the behaviour is as desired.
